connect/mirror-client/src/main/java/org/apache/kafka/connect/mirror/ReplicationPolicy.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.mirror;
+
+/** Defines which topics are "remote topics", e.g. "us-west.topic1". */
+public interface ReplicationPolicy {
```

Comment:
Looks like it's missing the methods like shouldCheckPointTopic. I'm assuming that is because this is still WiP.

Reply:
I plan to remove those methods from the KIP, for a few reasons:

1) the same logic is encoded in the config, for the most part. e.g. shouldCheckpointTopic is just combining the topic and group whitelists. If we already have those properties, it's best not to provide a second mechanism to redefine the behavior here. I think regexes are sufficiently powerful.

2) I want to enable using the same replication policy for both the connectors and the clients. The clients don't care about most of the methods in the KIP, so it's best not to make a client define them.

3) I want replication policies to be pretty much static across an entire organization, not per-cluster. The organization decides what remote topics look like, and then all connectors and clients know how to interpret them. So anything related to specific topics or groups doesn't fit that goal.

It's possible that ReplicationPolicy is no longer a good name for this, but I think it works.